crates/subspace-service/src/piece_cache.rs

```diff
@@ -120,17 +120,13 @@ where
             return Ok(());
         }
 
-        let insert_indexes = (u64::from(first_piece_index)..)
-            .take(pieces.len())
-            .map(PieceIndex::from)
-            .collect::<Vec<_>>();
+        let insert_indexes = (first_piece_index..).take(pieces.len()).collect::<Vec<_>>();
 
         let delete_indexes = u64::from(first_piece_index)
             .checked_sub(u64::from(self.max_pieces_in_cache))
+            .map(PieceIndex::from)
```

Comment:
Maybe it could be math on `PieceIndex`-es?

Reply:
We have math operations, but `derive_more` didn't have checked math derives and I considered that for a single use case (we have no other in the code base) it wasn't worth implementing. But yeah, it is not pretty :confused: